hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AMiscOperations.java

```diff
@@ -114,7 +114,7 @@ public void testPutObjectDirect() throws Throwable {
           new ByteArrayInputStream("PUT".getBytes()),
           metadata);
       LambdaTestUtils.intercept(IllegalStateException.class,
-          () -> fs.putObjectDirect(put, PutObjectOptions.keepingDirs()));
+          () -> fs.putObjectDirect(put, PutObjectOptions.keepingDirs(), null));
```

Comment:
Why not grab the duration tracker from the FS here?

```suggestion
          () -> fs.putObjectDirect(put, PutObjectOptions.keepingDirs(), fs.getDurationTrackerFactory()));
```

Maybe the question is when should we?

Reply:
i decided not to bother, as it does help check the null-value passthrough works

Reply:
can't argue with that :)